backends/saves/savefile.cpp

```diff
@@ -56,6 +56,16 @@ int32 OutSaveFile::pos() const {
 	return _wrapped->pos();
 }
 
+bool OutSaveFile::seek(int32 offset, int whence) {
+	SeekableWriteStream *ws = dynamic_cast<SeekableWriteStream *>(_wrapped);
+	return ws ? ws->seek(offset, whence) : false;
+}
+
+int32 OutSaveFile::size() const {
+	SeekableWriteStream *ws = dynamic_cast<SeekableWriteStream *>(_wrapped);
```

Comment:
There are (or at least, used to be) many savefile implementions which do not provide seeking. And as you point out, compressed saves don't support seeking.

So then engines expecting seekable savefiles won't work. But then the engine has no way to find out about this unless they try to seek here, it seems? Well, I guess they could try to seek and check for an error; but perhaps it would make more sense to add a specific method to check whether seeking is supported? 

But even then, you'd exclude many backends. Perhaps a better solution for your original problem would be to first create the savestate inside a memory buffer (possibly managed by a `SeekableMemoryWriteStream`), and then once that is done, just dump the whole thing into a normal `OutSaveFile`?

Reply:
I agree with @fingolfin comment that using first a seekable memory stream and then dumping it to a file in the engine might be a better approach than trying to make the file stream seekable.

Reply:
fingolfin: I wasn't aware that that would be the case (other backends not using StdIOStream). But in retrospect I guess it makes sense, since particularly for consoles, save file creation/retrieval could need special custom code. I guess having SeekableWriteStream for the Magnetic sub-engine is a bust.

The question remains whether I should revert OutSaveFile from SeekableWriteStream to WriteStream. I'm leaning towards yes, since it will avoid confusion in the future until and unless all supported save file implementations ever support it.

Reply:
@dreammaster: I would also lean toward reverting that change.

Reply:
And done. I've dropped the OutSaveFile commit.